dev/archery/archery/integration/datagen.py

```diff
@@ -927,6 +927,85 @@ class LargeListColumn(_BaseListColumn, _LargeOffsetsMixin):
     pass
 
 
+class ListViewField(Field):
+
+    def __init__(self, name, value_field, *, nullable=True,
+                 metadata=None):
+        super().__init__(name, nullable=nullable,
+                         metadata=metadata)
+        self.value_field = value_field
+
+    @property
+    def column_class(self):
+        return ListViewColumn
+
+    def _get_type(self):
+        return OrderedDict([
+            ('name', 'listview')
+        ])
+
+    def _get_children(self):
+        return [self.value_field.get_json()]
+
+    def generate_column(self, size, name=None):
+        MAX_LIST_SIZE = 4
+
+        is_valid = self._make_is_valid(size)
+        offsets = []
+        sizes = np.random.randint(0, MAX_LIST_SIZE + 1, size=size)
+        offset = 0
+        for s in sizes:
+            offsets.append(offset)
+            offset += int(s)
```

Comment:
Offsets should probably be out of order to break code that assumes list-view offsets are sorted like in lists.

Reply:
https://github.com/apache/arrow/pull/39019/commits/03a12f9cf50dbb92184afd815a73454b138ce713